libs/components/core/src/lib/modules/resize-observer/resize-observer-media-query.service.ts

```diff
@@ -122,11 +130,38 @@ export class SkyResizeObserverMediaQueryService implements OnDestroy {
       });
   }
 
-  #updateBreakpoint(breakpoint: SkyMediaBreakpoints) {
+  #updateBreakpoint(
+    breakpoint: SkyMediaBreakpoints,
+    updateResponsiveClasses?: boolean
+  ): void {
+    if (updateResponsiveClasses) {
+      this.#updateResponsiveClasses(this.#currentBreakpoint, breakpoint);
+    }
+
     this.#currentBreakpoint = breakpoint;
     this.#currentBreakpointObs.next(breakpoint);
   }
 
+  #updateResponsiveClasses(
+    oldBreakpoint: SkyMediaBreakpoints,
+    newBreakpoint: SkyMediaBreakpoints
+  ): void {
+    const oldClass = `sky-responsive-container-${
```

Comment:
These statements are practically identical except for which breakpoint you're looking for. Factor this out into a function. Example:

```
#getClassForBreakpoint(breakpointName: SkyMediaBreakpoints) {
  return `sky-responsive-container-${
      this.#breakpoints.find((breakpoint) => breakpoint.name === breakpointName)
        ?.classSuffix
    }`;
}
```

You may also want to reconsider always creating the CSS class because you have a `?.` in the statement, which means it's possible to end up with something like `sky-responsive-container-undefined`.